mountinfo/mounted_unix.go

```diff
@@ -32,13 +32,13 @@ func mountedByStat(path string) (bool, error) {
 
 func normalizePath(path string) (realPath string, err error) {
 	if realPath, err = filepath.Abs(path); err != nil {
-		return "", err
+		return path, err
 	}
 	if realPath, err = filepath.EvalSymlinks(realPath); err != nil {
-		return "", err
+		return path, err
 	}
 	if _, err := os.Stat(realPath); err != nil {
-		return "", err
+		return path, err
 	}
 	return realPath, nil
```

Comment:
Can `filepath.Abs()` or `filepath.EvalSymlinks()` also return an `os.ErrNotExist` error for other reasons? e.g. `filepath.Abs()` calls out to `os.GetCWD()`, which can return an error; https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/os/getwd.go#L22

And I assume `filepath.EvalSymlinks()`  can return an error if any symlink failed to be resolved.

Mostly wondering if `path` is "safe to consume" in all of these error situations. It may be a bit controversial to return _both_ an error _and_ a value. Not sure if there's problematic cases 🤔. 

Not sure what's best in that case though; should the caller use the "original" (non-normalised) path? (concerned if the "normalised" path in that case may contain some path that cannot / should not be used (could be an empty string returned by any of the above?)

Reply:
You are correct, it can fail. I thought about this a bit and made the following code changes.
When `normalizePath` fails or returns an error, we should return `"", err`. Now the only variable is the original path.

I think when ENOENT occurs, we should use the non-normalised path and read from /proc/mounts. This is because /proc/mounts will always have cannonicalized Path. These are the three cases I could think of:
1. path is a absolute path that returned ENOENT: The entry will not exist in /proc/mounts and return false
2. path is a symlink that returned ENOENT: The entry to the symlink will not exist in /proc/mounts and return false.
3. path is not-absolute: If filepath is not absolute, then we can try to get an Absolute filepath and check /proc/mounts. If an error is thrown, we can return it. 

wdyt? I made the changes though.